SimDataFormats/TrackingHit/interface/PSimHit.h

```diff
@@ -105,6 +107,15 @@ class PSimHit {
    */
   unsigned int trackId() const { return theTrackId; }
 
+  /** In case te SimTrack ID is incremented by the k_tidOffset for hit category definition, this
+   * methods returns the original theTrackId value directly.
+   */
+  unsigned int originalTrackId() const { return (theTrackId > k_tidOffset) ? theTrackId % k_tidOffset : theTrackId; }
+
+  unsigned int offsetTrackId() const { return (theTrackId > k_tidOffset) ? theTrackId / k_tidOffset : theTrackId; }
```

Comment:
@civanch sorry, but I see that there is a logical mistake here: in case no offset is added, this method should be logically return 0, not the trackid, do you agree? I will fix it, as the method is not used anywwhere so fare the impact of this is invisible in tests

Reply:
Yes, you are right, I overlook this. If you fix this, then there is a question: may be originalTrackId() is not needed, we may simply move this implementation to the method trackId()